### DISCUSSION TITLE
path does not work if there is a matchpath

### DISCUSSION THREAD

**Discussion by romeorows** [gatsbyjs/gatsby#33721] [Help]
When creating pages with createPage. Setting the path and matchPath with added parameters. I would expect both will resolve the same component layout and render the same content. In my case, only the matchPath is rendering components and the original path does not render any.

createPage({
     path: slug,
     matchPath: slugWithParameters, //Only this renders component
     component: layout,
     context: {
          id
     },
});

This is my reach Router:

<Router>
    <PrivateRoute page={Page} path={slugWithParameters} {...props} />
</Router>

Shouldn't the path and matchPath render the same content? Do I need to support the original path in the Router as well?

slug can be: "/", "/about", "/products"

slugWithParameters can be: "/:id", "/:id/about", "/:id/products"

**Comment by stevepepple** [comment]
I'm also running into issue with this. If I have a match path, the page context is incorrect. If I remove the match path, the content is right, but then I must parse the path on the client and map the path params. 

If Gatsby could provide example of best practices for nested routes, that would be very helpful. I've spent quite a bit of time fussing with this API because it's not intuitive.

**Reply by stevepepple** [reply]
I'll also elaborate and say that I have pages nested in these path using the createPages API. I'm going to create static path for most of the routes, but also want to handle dynamic client routes. 

Example Routes: 
```
/cities/
/cities/:city/
/cities/:city/:activity_category
```
